feat: Create sharing menu

john0isaac opened the pull request:
closes #255 
- Import Material Menu Library.
- Replaced sharing button with sharing Menu.
- Added a string parameter to track the clicked button.
- Redefined Sharing Function.
- Included if statements to check the clicked button and configure sharing accordingly.

While working on this I found a new issue the hashtags weren't added correctly to the twitter sharing link.
I fixed this issue now if you try to share on twitter the hashtags will appear.

Adding more platforms now will be much easier as it's only a copy-and-paste process. (see [this](https://dev.to/shahednasser/how-to-easily-add-share-links-for-each-social-media-platform-1l4f))
![image](https://github.com/Azure-Samples/contoso-real-estate/assets/64026625/c69b39e8-55fc-4c92-ae3e-5460d97ff961)

anfibiacreativa commented on packages/portal/src/app/rentalpage/rentalpage.component.ts:
```diff
@@ -1,6 +1,7 @@
 import { CommonModule } from "@angular/common";
 import { Component, OnInit, inject, signal } from "@angular/core";
 import { ActivatedRoute, Navigation, Router } from "@angular/router";
+import {MatMenuModule} from '@angular/material/menu';
```
To comply with project conventions, please add a space between curly brackets and import names. Thank you!

john0isaac replied:
@anfibiacreativa done, thank you for reviewing!

anfibiacreativa commented on packages/portal/src/app/shared/listing.service.ts:
```diff
@@ -39,16 +39,27 @@ export class ListingService {
     const rent = listing.fees.at(0);
     const currency = listing.fees.at(-1);
-    this.windowService
+    if(platform === "facebook"){
```
If we wanted to conform with conventions, there should be a space between the `if` clause and the statement parenthesis. 
We may want here to test if platform is defined at all 

`if (platform && platform === `

But I think we could consider assigning the url value to a conditional value called for example shareUrl. The way I would do it is I would create a global object that has the twitter and facebook urls as properties (so you can extend to more should you need) . 

```
const socialMediaUrls = {
    twitter = '[url-value]',
    facebook = '[url-value]
 ;
```

Then you can import and deconstruct, and execute the conditional for the url value only. // (Sort of quick pseudo code warning here, but you'll get the idea ;) )
```
const { twitter, facebook } = socialMediaUrls;
const shareUrl = (platform) => { platform === 'twitter' ? twitter : facebook; }
```

Then we'd only need to pass shareUrl to the same `this.windowService.native().open(` function.
I see you're also passing the url further down the line and the param name is &u= in one case and &url= in the other. We could also assign that segment of the url to a variable and do the same kind of conditional test to determine what's the corresponding value.

john0isaac replied:
@anfibiacreativa Thank you for your amazing feedback I learned a lot!

john0isaac commented:
- Added Spaces around imports, if statement.
- Added check for the platform value definition.
- Destructure the code for sharing URL, URL of the page.

I couldn't figure out the part where you mentioned creating a **global object** for the sharing URL I did it inside the function as I didn't know if there was a file to create the global variables in or if should I just have included them after the imports in the same file.
I did create an object though inside the share function.

As for the part where you recommend using the **ternary operator** I think it will get more complicated as we add more social media platforms so, I wouldn't recommend it although it's a beautiful idea if we only had two platforms.

Thanks again for the amazing feedback.

sinedied reviewed (approved):
Just added a small suggestion, as Twitter has been renamed to X we might as well do so here 😉

john0isaac commented:
@sinedied I approved your changes.
Thank you for reviewing!
You can merge now.

anfibiacreativa reviewed (approved):
Thanks for your contribution!

anfibiacreativa reviewed (approved):
Thanks for your contribution!
